libxrdp/xrdp_orders_rail.c

```diff
@@ -195,15 +195,21 @@ xrdp_orders_send_as_unicode(struct stream *s, const char *text)
     int str_chars;
     int index;
     int i32;
-    twchar wdst[256];
+    int len;
+    twchar *wdst;
 
-    str_chars = g_mbstowcs(wdst, text, 255);
+    len = g_strlen(text) + 1;
 
+    wdst = (twchar *) g_malloc(sizeof(twchar) * len, 1);
+    if (wdst == 0)
+    {
+        return 1;
+    }
+    str_chars = g_mbstowcs(wdst, text, sizeof(twchar) * len);
```

Comment:
The last argument to mbstowcs and g_mbstowcs is the number of wide characters, not bytes. So it should be `len`, not `sizeof(twchar) * len`. We are promising more space than what we have.